test/utils/Base64.t.sol

```diff
@@ -6,27 +6,33 @@ import {Test} from "forge-std/Test.sol";
 
 import {Base64} from "@openzeppelin/contracts/utils/Base64.sol";
 
-/// NOTE: This test requires `ffi` to be enabled. It does not run in the CI
-/// environment given `ffi` is not recommended.
-/// See: https://github.com/foundry-rs/foundry/issues/6744
 contract Base64Test is Test {
     function testEncode(bytes memory input) external {
         string memory output = Base64.encode(input);
-        assertEq(output, _base64Ffi(input, "encode"));
+        assertEq(output, vm.toBase64(input));
     }
 
     function testEncodeURL(bytes memory input) external {
         string memory output = Base64.encodeURL(input);
-        assertEq(output, _base64Ffi(input, "encodeURL"));
+        assertEq(output, _removePadding(vm.toBase64URL(input)));
     }
 
-    function _base64Ffi(bytes memory input, string memory fn) internal returns (string memory) {
-        string[] memory command = new string[](4);
-        command[0] = "bash";
-        command[1] = "scripts/tests/base64.sh";
-        command[2] = fn;
-        command[3] = vm.toString(input);
-        bytes memory retData = vm.ffi(command);
-        return string(retData);
+    function _removePadding(string memory input) internal pure returns (string memory) {
+        bytes memory bytesInput = bytes(input);
+        uint256 length = bytesInput.length;
+        if (length == 0) return input;
+
+        uint256 padding = 0;
+
+        while (bytesInput[length - padding - 1] == 0x3d) {
+            padding++;
+        }
```

Comment:
The fact that the while does not overflow only depends on the string not boing only paddings.

Maybe we should use a for to explicit the end.
```suggestion
        for (padding=0; padding<length; ++padding) {
            if (bytesInput[length - padding - 1] != 0x3d) break;
        }
```

Reply:
That way we should be able to remove the `if (length == 0) return input;` check